src/DebugEngineHost/HostNatvisProject.cs

```diff
@@ -73,6 +81,74 @@ internal enum VSENUMPROJFLAGS
                 //  file as well as all virtual projects including nested (a.k.a. sub) projects
             };
 
+            /// <summary>
+            /// Gets the WorkspaceService from Microsoft.VisualStudio.Workspace
+            /// </summary>
+            /// <remarks>This package won't be automatically loaded by MEF, so we need to manually acquire exported MEF Parts.</remarks>
+            private static IVsFolderWorkspaceService GetWorkspaceService()
+            {
+                IComponentModel componentModel = ServiceProvider.GlobalProvider.GetService(typeof(SComponentModel).GUID) as IComponentModel;
+                var workspaceServices = componentModel.DefaultExportProvider.GetExports<IVsFolderWorkspaceService>();
+
+                if (workspaceServices != null && workspaceServices.Count() > 0)
+                {
+                    return workspaceServices.First().Value;
+                }
+                return null;
+            }
+
+            private static int GetOpenFolderSourceLocations(string bstrFileName, out bool isIndexComplete, out IEnumerable<string> pSourcesArray)
+            {
+                isIndexComplete = false;
+                var workspaceService = GetWorkspaceService();
+
+                if (workspaceService != null)
+                {
+                    IWorkspace currentWorkspace = workspaceService.CurrentWorkspace;
+                    IIndexWorkspaceService indexWorkspaceService = currentWorkspace?.GetService<IIndexWorkspaceService>(throwIfNotFound: false);
+                    if (indexWorkspaceService != null)
+                    {
+                        isIndexComplete = indexWorkspaceService.State == IndexWorkspaceState.Completed;
```

Comment:
If the index is not up to date, do we want to log a warning to the output window?